base/strings/string.jl

```diff
@@ -92,7 +92,7 @@ String(s::CodeUnits{UInt8,String}) = s.s
 ## low-level functions ##
 
 pointer(s::String) = unsafe_convert(Ptr{UInt8}, s)
-pointer(s::String, i::Integer) = pointer(s)+(i-1)
+pointer(s::String, i::Integer) = pointer(s) + (i - 1)::Integer
```

Comment:
Sometimes these are written as `Int(i)::Int - 1`.

Reply:
Is there any concern about having values too large to convert to an `Int`. It seems unlikely that would occur but I figured I'd ask

Reply:
I'd think string length is pretty safe; someone would need to have a single string that's more than half the size of memory on a 32-bit machine. I guess you could alternatively use `UInt`.